code/src/components/Welcome.js

```diff
@@ -0,0 +1,39 @@
+import React from "react";
+import { useSelector } from "react-redux";
+import styled from "styled-components/macro";
+
+import { UserInput } from "./UserInput";
+import { Labyrinth } from "./Labyrinth";
+
+export const Welcome = () => {
+  const currentGameState = useSelector(store => store.game.currentGameState);
+
+  if (!currentGameState.gameData) {
+    return (
+      <MainWrapper>
+        <UserInput />
+        <Image src="./assets/labyrinth-poster.jpg" alt="labyrinth" />
+      </MainWrapper>
+    );
+  } else {
+    return <Labyrinth />;
+  }
+};
```

Comment:
Could be written with ternary operators and then you don't need to use return twice.